From 8c7bd35db47726af4be5d2e5364aeace053ab1f0 Mon Sep 17 00:00:00 2001
From: Shunqian Zheng <zhengsq@rock-chips.com>
Date: Wed, 27 Jun 2018 09:40:05 +0800
Subject: [PATCH] ARM64: dts: rockchip: add firefly ov13850 camera

Add firefly linux ov13850 camera node, and enable isp
and isp_mmu.

Change-Id: I317f4b9895c73913456c957273cc6b18173a8cd9
Signed-off-by: Shunqian Zheng <zhengsq@rock-chips.com>
---
 .../dts/rockchip/rk3399-firefly-linux.dts     | 46 +++++++++++++++++++
 1 file changed, 46 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
index 243eef889e5f..c3e076f7ff89 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
@@ -670,6 +670,43 @@
 		io-channels = <&saradc 4>;
 		hp-det-adc-value = <500>;
 	};
+
+	camera0: ov13850@10 {
+		status = "okay";
+		compatible = "omnivision,ov13850-v4l2-i2c-subdev";
+		reg = < 0x10 >;
+		device_type = "v4l2-i2c-subdev";
+
+		clocks = <&cru SCLK_CIF_OUT>;
+		clock-names = "clk_cif_out";
+
+		pinctrl-names = "rockchip,camera_default", "rockchip,camera_sleep";
+		pinctrl-0 = <&cam0_default_pins>;
+		pinctrl-1 = <&cam0_sleep_pins>;
+
+		rockchip,pd-gpio = <&gpio2 12 GPIO_ACTIVE_LOW>;
+		rockchip,pwr-gpio = <&gpio1 23 GPIO_ACTIVE_HIGH>;
+		rockchip,pwr-2nd-gpio = <&gpio1 22 GPIO_ACTIVE_HIGH>;
+		rockchip,rst-gpio = <&gpio0 8 GPIO_ACTIVE_LOW>;
+
+		rockchip,camera-module-mclk-name = "clk_cif_out";
+		rockchip,camera-module-facing = "back";
+		rockchip,camera-module-name = "cmk-cb0695-fv1";
+		rockchip,camera-module-len-name = "lg9569a2";
+		rockchip,camera-module-fov-h = "66.0";
+		rockchip,camera-module-fov-v = "50.1";
+		rockchip,camera-module-orientation = <0>;
+		rockchip,camera-module-iq-flip = <0>;
+		rockchip,camera-module-iq-mirror = <0>;
+		rockchip,camera-module-flip = <1>;
+		rockchip,camera-module-mirror = <0>;
+
+		rockchip,camera-module-defrect0 = <2112 1568 0 0 2112 1568>;
+		rockchip,camera-module-defrect1 = <4224 3136 0 0 4224 3136>;
+		rockchip,camera-module-defrect3 = <3264 2448 0 0 3264 2448>;
+		rockchip,camera-module-flash-support = <0>;
+		rockchip,camera-module-mipi-dphy-index = <0>;
+	};
 };
 
 &i2c3 {
@@ -1016,6 +1053,15 @@
 	status = "okay";
 };
 
+&cif_isp0 {
+	rockchip,camera-modules-attached = <&camera0>;
+	status = "okay";
+};
+
+&isp0_mmu {
+	status = "okay";
+};
+
 &vopb {
 	status = "okay";
 };
-- 
2.35.3

